src/components/Input/types.ts

```diff
@@ -42,5 +47,43 @@ export interface InputProps {
     /**
      * the input can use state to react to user interaction
      */
-    state?: 'error' | 'confirmed';
+    state?: 'error' | 'confirmed' | 'disabled';
```

Comment:
I like this because we have state, but also `disabled` is a property that inputs use. So I wonder should we just use the standard HTML way like with Button? What do you think? Josef was keen on keeping as much props as standard when we started, like the onClick and onChange events too are standard HTML props

Reply:
I started by adding the independent prop `disabled`. So it was possible to use it like this: <Input ... disabled />. But since we have the `state` prop which will be 100% used for doing custom logic like:
```
let inputState;
if (!hasAccess) inputState = "disabled"
if (dataIsValid) inputState = "confirmed"
<Input ... state={inputState} />
```

If there will be separated props and the user write something like `<Input error={true} confirmed={true} disabled={true} />`  we should work on logic like it's not allowed for example to have `confirmed` and `error` props `true`

Reply:
or we can leave the state and add `disabled` only as independent

Reply:
If the second option - I like it

Reply:
i just think its slightly better DX to use the props the dev will expect rather than hiding it in state. But if you want to use state for logic that cool.
`<Input disbaled />`
to be is more intuitive than
z<Input state="disabled" />`

Reply:
<Input disabled /> is amazing